ui/packages/catalog-ui-search/src/main/webapp/component/search-form-interactions/search-form-interactions.view.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+/*global define, window*/
+
+
+
+define([
+    'wreqr',
+    'marionette',
+    'underscore',
+    'jquery',
+    './search-form-interactions.hbs',
+    'js/CustomElements',
+    'js/store',
+    'component/router/router',
+    'component/singletons/user-instance',
+    'component/loading/loading.view',
+    'component/lightbox/lightbox.view.instance',
+    'component/announcement',
+    'component/confirmation/confirmation.view'
+], function(wreqr, Marionette, _, $, template, CustomElements, store, router, user,
+    LoadingView, lightboxInstance, announcement, ConfirmationView) {
+    return Marionette.ItemView.extend({
+        template: template,
+        tagName: CustomElements.register('search-form-interactions'),
+        className: 'composed-menu',
+        modelEvents: {
+            'change': 'render'
+        },
+        events: {
+            'click .interaction-trash': 'handleTrash',
+            'click .search-form-interaction': 'handleClick'
+        },
+        ui: {},
+        initialize: function() {},
+        onRender: function() {
+            this.checkIfSubscribed();
+            //this.handleLocal();
+        },
+        handleLocal: function() {
+            this.$el.toggleClass('is-local', this.model.isLocal());
+        },
+        checkIfSubscribed: function() {
+            this.$el.toggleClass('is-subscribed', Boolean(this.model.get('subscribed')));
+        },
+        handleTrash: function() {
+            var loginUser = user.get('user');
+            if(loginUser.attributes.username === this.model.attributes.createdBy)
+            {
+                this.listenTo(ConfirmationView.generateConfirmation({
+                    prompt: 'This will permanently delete the template. Are you sure? ',
+                    no: 'Cancel',
+                    yes: 'Delete'
+                }),
+                'change:choice',
+                function(confirmation) {
+                    if (confirmation.get('choice')) {
+                        var loadingview = new LoadingView();
+                        this.model.url = '/search/catalog/internal/forms/' + this.model.id;
+                        this.model.destroy();
```

Comment:
Backbone does an optimistic destroy. This means items from the collection will get removed from the collection regardless of the response (500 vs 200). So add the parameters {wait: true} to this.model.destroy() and it will wait for the server response.

Reply:
Added wait